### PR TITLE
Pipeline with no DPUs cannot be saved, runtime reports failure event

### LINKED ISSUE #410: Blank pipeline run ends with FAILED status without any error message
When blank pipeline is saved and run, it ends with FAILED status but no error message explaining the reason is displayed;
This is very confusing; 
I would expect it to fail with message that no DPU found to execute;

Another possibility would be mark pipeline status as WARNING and give proper message that pipeline has no DPUs

Motivation:
- this bug was discovered when updating DPUs via UV master API.
- all DPU instances were deleted, so blank pipelines remained (also a bug, but still can happen something like this)
- pipelines started to fail without any error message

### PR DESCRIPTION
**Pull request by mkl-eea-sk**
Resolves #410

### CODE REVIEW

File: backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/execution/pipeline/Executor.java [hunk 1]
```diff
@@ -286,7 +286,10 @@ private void execute() {
 
         // we need result state
         ExecutionResult execResult = new ExecutionResult();
-
+        if (!dependencyGraph.iterator().hasNext()) {
+            eventPublisher.publishEvent(PipelineFailedEvent.create("Blank pipeline with no DPUs cannot be executed", "Pipeline contains no DPUs, cannot be executed", null, this.execution, this));
```
Comment (mkl-eea-sk) [inline backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/execution/pipeline/Executor.java]: Localize this message!

**Comment by mkl-eea-sk** [comment]
before merge localize: https://github.com/UnifiedViews/Core/pull/451/files#r34864923

File: commons-app/src/main/java/cz/cuni/mff/xrg/odcs/commons/app/facade/PipelineFacadeImpl.java [hunk 2]
```diff
@@ -166,7 +165,10 @@ public Pipeline getPipeline(long id) {
             + "AND hasRole('pipeline.setVisibilityPublicRw')))")
     @Override
     public void save(Pipeline pipeline) {
-
+        if (pipeline.getGraph().getNodes().isEmpty()) {
```
Comment (mkl-eea-sk) [inline commons-app/src/main/java/cz/cuni/mff/xrg/odcs/commons/app/facade/PipelineFacadeImpl.java]: remove this check (breaks master api)

File: frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/PipelineEdit.java [hunk 3]
```diff
@@ -968,12 +973,16 @@ private void setupButtons(boolean isModified) {
...
 
-    private void savePipelineAsNew() {
+    private boolean savePipelineAsNew() {
         if (!pipelineFacade.isUpToDate(pipeline)) {
         }
         if (!validate()) {
-            return;
+            return false;
         }
+        if (!pipelineCanvas.validateGraph()) {
```
Comment (tomas-knap) [inline frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/PipelineEdit.java]: This test on validity of graph should be removed, it is too restrictive. It must work in a way that "If I click validate, I got issues". Nevertheless it must be possible to save pipeline which is not 100% valid (as it was working until now) Motivation: Sometimes, preparing pipeline is matter of hours, days, so I cannot be restricted by being able to store only valid pipelines. Also when debugging, the pipeline may not need to be 100% valid.

File: frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/PipelineEdit.java [hunk 4]
```diff
@@ -1178,6 +1188,9 @@ protected boolean savePipeline(final String successAction) {
         if (!validate()) {
             return false;
         }
+        if (!pipelineCanvas.validateGraph()) {
```
Comment (tomas-knap) [inline frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/PipelineEdit.java]: See above, test on validity of graphs must be removed

Comment (mkl-eea-sk) [inline frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/PipelineEdit.java]: well, so whole requirements on 'lets do not save blank pipeline' can be dropped too?

Comment (tomas-knap) [inline frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/PipelineEdit.java]: Yes, it should be possible to save empty pipeline. But when running such pipeline better explanation of the failure should be provided

**Pull request merged**